mock/mock.go

```diff
@@ -526,9 +519,9 @@ func AssertExpectationsForObjects(t TestingT, testObjects ...interface{}) bool {
 		h.Helper()
 	}
 	for _, obj := range testObjects {
-		if m, ok := obj.(Mock); ok {
+		if m, ok := obj.(*Mock); ok {
```

Comment:
Are you sure this should be changed? IIRC I didn't change this logic in the PR I merged?

Reply:
Yes, because we want to refer to the same instance of sync.Mutex for the copy that is a type check here.

Reply:
Also, `go vet ./...` will show that warning for us ( I think it was caught in CI).

Reply:
Fair enough; I see it's a test helper anyways.

Thank you for catching this!

Reply:
🤦 . I must've missed this when doing the Go 1.18 PR (which broke this) and 'fixed it' the long way around. Thanks